compiler+runtime/test/cpp/jank/runtime/core.cpp

```diff
@@ -8,6 +9,20 @@ namespace jank::runtime
 {
   TEST_SUITE("core runtime")
   {
+    TEST_CASE("subs")
+    {
+      auto const s{ make_box<obj::persistent_string>("foo bar") };
+      CHECK(equal(subs(s, make_box(0)), make_box<obj::persistent_string>("foo bar")));
+      CHECK(equal(subs(s, make_box(1)), make_box<obj::persistent_string>("oo bar")));
+      CHECK(equal(subs(s, make_box(2)), make_box<obj::persistent_string>("o bar")));
+      CHECK(equal(subs(s, make_box(0), make_box(7)), make_box<obj::persistent_string>("foo bar")));
+      CHECK(equal(subs(s, make_box(0), make_box(0)), make_box<obj::persistent_string>("")));
+      CHECK(equal(subs(s, make_box(1), make_box(1)), make_box<obj::persistent_string>("")));
+      CHECK(equal(subs(s, make_box(1), make_box(6)), make_box<obj::persistent_string>("oo ba")));
+      CHECK(equal(subs(s, make_box(1), make_box(7)), make_box<obj::persistent_string>("oo bar")));
+      CHECK(equal(subs(s, make_box(3), make_box(4)), make_box<obj::persistent_string>(" ")));
+    }
+
```

Comment:
This is duplicating tests coverage from clojure-test-suite. We don't want to do that.

I accepted the `equal` tests below because they use `nullptr`, which is not currently possible to represent from jank code itself, so this had to be done in C++.

I know we can't run the `subs` tests in the clojure-test-suite. We'll get there, but we don't want to duplicate test coverage along the way.

Reply:
Removed the tests.